docs/root/start/sandboxes/tls-sni.rst

```diff
@@ -0,0 +1,169 @@
+.. _install_sandboxes_tls_sni:
+
+TLS Server name indication (``SNI``)
+====================================
+
+.. sidebar:: Requirements
+
+   .. include:: _include/docker-env-setup-link.rst
+
+   :ref:`curl <start_sandboxes_setup_curl>`
+	Used to make ``HTTP`` requests.
+
+   :ref:`jq <start_sandboxes_setup_jq>`
+	Parse ``json`` output from the upstream echo servers.
+
+This example demonstrates an Envoy proxy that listens on multiple domains
+on the same ``IP`` address and either provides separate ``TLS`` termination
+or proxies to an upstream ``TLS`` service for each.
```

Comment:
Maybe note which ones are which of domain1, domain2, and domain3, so the reader can refer to the config and understand what's going on.